modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -177,41 +132,66 @@ public Slot getSlot(String queueName, String nodeId) {
      * @return slot object
      */
     private Slot getFreshSlot(String queueName) {
+
         Slot slotToBeAssigned = null;
-        TreeSetLongWrapper wrapper = slotIDMap.get(queueName);
-        TreeSet<Long> messageIDSet;
-        if (null != wrapper) {
-            messageIDSet = wrapper.getLongTreeSet();
-            if (messageIDSet != null && !messageIDSet.isEmpty()) {
-
-                //create a new slot
-                slotToBeAssigned = new Slot();
-
-                //start msgID will be last assigned ID + 1 so that slots are created with no
-                // message ID gaps in-between
-                Long lastAssignedId = queueToLastAssignedIDMap.get(queueName);
-                if (lastAssignedId != null) {
-                    slotToBeAssigned.setStartMessageId(lastAssignedId + 1);
-                } else {
-                    slotToBeAssigned.setStartMessageId(0L);
-                }
+        TreeSet<Long> messageIDSet = null;
+
+        // Get message id set from database
+        try {
+            messageIDSet = rdbmsSlotAgent.getMessageIds(queueName);
+        } catch (AndesException e){
+            log.error("Failed to get message ids for queue: " + queueName);
+        }
+
+        if (messageIDSet != null && !messageIDSet.isEmpty()) {
+
+            slotToBeAssigned = new Slot();
+
+            //start msgID will be last assigned ID + 1 so that slots are created with no
+            // message ID gaps in-between
+            long lastAssignedId = 0L;
+
+            try {
+                lastAssignedId = rdbmsSlotAgent.getQueueToLastAssignedId(queueName);
+            } catch (AndesException ex) {
+                log.error("Failed to get last assigned id for queue: " + queueName);
+            }
```

Comment:
Yes. If failed this method will return a null object which will be handled by the caller. So it is necessary to continue.